sdk/python/feast/infra/offline_stores/bigquery_source.py

```diff
@@ -64,7 +64,7 @@ def __init__(
             else:
                 warnings.warn(
                     (
-                        "Starting in Feast 0.21, Feast will require either a name for a data source (if using query) or `table`."
+                        f"Starting in Feast 0.21, Feast will require either a name for a data source (if using query) or `table`: {self.query}"
```

Comment:
We should print out `self.query` only if it's set here

Reply:
if it hits this it should already have query defined